IPython/sphinxext/ipython_directive.py

```diff
@@ -790,74 +790,56 @@ def process_pure_python(self, content):
         the content as a list as if it were ipython code
         """
         output = []
-        savefig = False # keep up with this to clear figure
-        multiline = False # to handle line continuation
-        multiline_start = None
         fmtin = self.promptin
-
         ct = 0
 
-        for lineno, line in enumerate(content):
-
-            line_stripped = line.strip()
-            if not len(line):
-                output.append(line)
-                continue
-
-            # handle decorators
-            if line_stripped.startswith('@'):
-                output.extend([line])
-                if 'savefig' in line:
-                    savefig = True # and need to clear figure
+        try:
+            savefigs = [i for i, line in enumerate(content)
+                        if line[:8] == '@savefig']
```

Comment:
I'm going to drop comment also for myself while reviewing and trying to understand what this code does. 

Is there anything that would prevent a `@savefig` to be indented ? Like in a loop or an if statement ?